src/pages/FrontPage.js

```diff
@@ -1,11 +1,214 @@
 import React from 'react';
 import CourseList from '../components/CourseList';
+import LessonFilter from '../components/filter/LessonFilter';
+
+const iconContext = require.context('lessonSrc/', true, /^\.\/[^\/]*\/logo-black\.png/);
+const lessonContext = require.context('onlyFrontmatter!lessonSrc/', true,
+  /^\.\/[^\/]*\/[^\/]*\/(?!README\.md$)[^\/]*\.md/);
 
 const FrontPage = React.createClass({
+  getInitialState() {
+    return {
+      filter: [],
+      tagGroups: [],
+      allCourses: [],
+      filteredCourses: []
+    };
+  },
+  componentWillMount() {
```

Comment:
Jeg vet det er fristende å lage `courses` og `tagGroups` i samme funksjon, siden du bruker litt av den samme infoen når du gjøre paths.forEach(), og du tenker at det er sikkert raskere å lage begge samtidig.

Min anbefaling er at du unngår denne fristelsen, for dette er optimalisering på bekostning av kompleksitet, og selv om denne optimaliseringen kan virke lur, så blir altså koden mer kompleks, og hvis du skiller ut funksjonaliteten her i to funksjoner getCourses() og getTagGroups() (som kanskje begge tar et `paths`-argument?), så vil du kanskje se at det ikke blir så ille likevel? Det gjør det ihvertfall lettere å forstå koden, og det blir lettere å skrive enhetstester.

Forstod du hva jeg mente nå, eller var jeg litt tåkete?